public/scripts/last.js

```diff
@@ -104,9 +100,22 @@ function getFullHistory(requestLength, previousDate, todayDate) {
       type:'POST',
       url: 'http://ws.audioscrobbler.com/2.0/?method=user.getRecentTracks&api_key=100a45f60fce336c43b1dac55062e23a&username=' + userLastId + '&from='+ previousDate +'&to='+ todayDate +'&page='+ i +'&format=json',
       success: function(response) {
-        for (i = 0; i <= (response.recenttracks.track.length - 1); i ++){
-          allCallSongs.push(response.recenttracks.track[i]);
+
+        for (j = 0; j <= (response.recenttracks.track.length - 1); j ++){
+          allCallSongs.push(response.recenttracks.track[j]);
         }
+
+      // NOTE: if one of these requests fails, then allCallSongs.length will never equal numTracksExpected
+      // and this if-statement will never be satisfied and the app will hang.
+      // Some error handling (maybe decrement `numTracksExpected` by `tracksPerPage` or somethhing in the error function?)
+      // could fix this potential problem.
+      if (allCallSongs.length >= numTracksExpected) {
+        console.log("last.fm history responses all received, now let's fetch spotify data!");
```

Comment:
left these console.logs in here in case they're useful for you, obviously feel free to delete them!